CRM/Contribute/BAO/Contribution.php

```diff
@@ -5669,4 +5674,79 @@ public static function replaceContributionTokens(
     return $contributionDetails;
   }
 
+  /**
+   * Record adjusted amount.
+   *
+   * @param int $updatedAmount
+   * @param int $paidAmount
+   * @param int $contributionId
+   *
+   * @param int $taxAmount
+   * @param bool $updateAmountLevel
+   *
+   * @return bool|\CRM_Core_BAO_FinancialTrxn
+   */
+  public static function recordAdjustedAmt($updatedAmount, $paidAmount, $contributionId, $taxAmount = NULL, $updateAmountLevel = NULL) {
```

Comment:
Or what if I move that function to same LineItem BAO file where it's ONLY been called and then I'll change the accessibility to private protected fn.